platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java

```diff
@@ -85,11 +87,12 @@ private static IncrementalInputProperties createIncrementalInputProperties(UnitO
             case NON_INCREMENTAL:
                 return IncrementalInputProperties.NONE;
             case INCREMENTAL:
+                Set<String> alreadyVisitedProperty = new HashSet<>();
                 ImmutableBiMap.Builder<String, Object> builder = ImmutableBiMap.builder();
                 InputVisitor visitor = new InputVisitor() {
                     @Override
                     public void visitInputFileProperty(String propertyName, InputBehavior behavior, InputFileValueSupplier valueSupplier) {
-                        if (behavior.shouldTrackChanges()) {
+                        if (behavior.shouldTrackChanges() && alreadyVisitedProperty.add(propertyName)) {
```

Comment:
Interesting, I though we already handled that here: https://github.com/gradle/gradle/blob/1d48a3a847929a832e89ef8c7a366d98497dc389/platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/impl/DefaultInputFingerprinter.java#L119-L124

Though it might be a different case in the resolve changes. Maybe the execution should make sure that you only provide one property in either `identityInputs` or `regularInputs`?